src/components/Pages/AvatarSetupPage/AvatarSetupPage.tsx

```diff
@@ -96,24 +97,12 @@ const AvatarSetupPage: React.FC = () => {
 
   const characterCount = useMemo(() => state.username.length, [state.username])
 
-  const hasError = useMemo(() => characterCount > MAX_CHARACTERS, [characterCount])
-
-  const nameError = useMemo(() => {
-    if (!state.username.length) {
-      return 'Please enter your username.'
-    }
-    if (state.username.length >= 15) {
-      return 'Sorry, usernames can have a maximum of 15 characters.'
-    }
-    if (state.username.includes(' ')) {
-      return 'Sorry, spaces are not permitted.'
-    }
-    if (!/^[a-zA-Z0-9]+$/.test(state.username)) {
-      return 'Sorry, special characters (!@#$%) are not permitted.'
-    }
-    return ''
+  const isUsernameValid = useMemo(() => {
+    return /^[a-zA-Z0-9]*$/.test(state.username)
   }, [state.username])
 
+  const hasUsernameCharacterCount = useMemo(() => characterCount > MAX_CHARACTERS || !isUsernameValid, [characterCount, isUsernameValid])
```

Comment:
WDYT about using the word correct or valid as the other variables?
```suggestion
  const hasCorrectUsernameCharacterCount = useMemo(() => characterCount > MAX_CHARACTERS || !isUsernameValid, [characterCount, isUsernameValid])
```
or
```suggestion
  const hasValidUsernameCharacterCount = useMemo(() => characterCount > MAX_CHARACTERS || !isUsernameValid, [characterCount, isUsernameValid])
```